### ISSUE TITLE
Provide a mechanism to select the current DSL

### ISSUE THREAD

**Issue by lordrip** [KaotoIO/kaoto-next#136]
### Context
Currently, there's no global state nor provider to determine the current DSL, and this is required to properly set the correct schema into the Source Code editor

### Goal
We need to decide whether to use a global UI store or a react provider to store the following information:
```
{
  currentDsl: CamelRoute | KameletBinding | Kamelet | Pipe | Integration
}
```

whereas each DSL object would have:
```
{
  name: string;
  schema: Record<string, unknown>;
  multipleRoute: boolean;
}
```

### Note
1. It would be beneficial to have a registry class to place the available DSL with their corresponding schemas and potentially allowed nodes (Camel Components, Processors, Kamelets, etc...)
2. This issue is a prerequisite of: https://github.com/KaotoIO/kaoto-next/issues/81

**Comment by lordrip** [comment]
After discussing it, it makes more sense to have a dedicated property within the EntityContext object in order to be determined when we're parsing the YAML source code.

**Comment by lhein** [comment]
I am quite curious why the source code editor cannot auto detect what is in the code. Of course you will have to provide a selector if you start with a blank source editor.

**Comment by lordrip** [comment]
There are two ways to signal the source code editor which schema or language to use:

1. Set all the schemas beforehand and instruct the source code editor to apply the correct one depending on the filename. We're doing this for `.camel.yaml` files in `VSCode`
2. Inspect what's the content of the source code editor (parsing) and apply the correct schema afterward.

Since we don't have filenames in `kaoto-next` at the moment, we'll go with the second approach, detecting the content and reacting accordingly.

**Comment by lordrip** [comment]
Closed by #172
